src/debugCodeLensProvider.ts

```diff
@@ -169,9 +170,35 @@ export async function startDebugging(mainClass: string, projectName: string, uri
     const workspaceFolder: vscode.WorkspaceFolder = vscode.workspace.getWorkspaceFolder(uri);
     const workspaceUri: vscode.Uri = workspaceFolder ? workspaceFolder.uri : undefined;
 
+    if (workspaceUri && !(await isOnClasspath(uri.toString())) && !(await addToClasspath(uri))) {
+        return false;
+    }
+
     const debugConfig: vscode.DebugConfiguration = await constructDebugConfig(mainClass, projectName, workspaceUri);
     debugConfig.projectName = projectName;
     debugConfig.noDebug = noDebug;
 
     return vscode.debug.startDebugging(workspaceFolder, debugConfig);
 }
+
+async function addToClasspath(uri: vscode.Uri): Promise<boolean> {
+    const fileName = path.basename(uri.fsPath || "");
+    const parentFsPath = path.dirname(uri.fsPath || "");
+    if (!parentFsPath) {
+        return true;
+    }
+
+    const parentUri = vscode.Uri.file(parentFsPath);
+    let parentPath = vscode.workspace.asRelativePath(parentUri, true);
+    if (parentPath === parentUri.fsPath) {
+        parentPath = path.basename(parentFsPath);
+    }
+    const ans = await vscode.window.showWarningMessage(`The file ${fileName} isn't on the classpath, the runtime may throw class not found error. `
```

Comment:
"in" or "on" the classpath?

Reply:
JDT api called it `isOnClasspath`, i would to keep it consistent.